rust/openvasd/src/controller/entry.rs

```diff
@@ -250,7 +251,13 @@ where
             response_blocking(move || {
                 let scans = ctx.scans.read()?;
                 match scans.get(&id) {
-                    Some(prgrs) => Ok(ctx.response.ok(&prgrs.scan)),
+                    Some(prgrs) => {
+                        // Clone the scan to remove sensitive data before
+                        // sending it back into the response
+                        let mut prgrs_aux = prgrs.clone();
+                        prgrs_aux.clean_sensitive_data();
```

Comment:
Do we consider everything in a credential confidential? With that a customer will never be aware if a scan even contains credentials which will be very confusing if a customer uses this endpoint to verify the creation of a scan. I think a replacement with `*` on passwords etc will be less confusing than just deletion.

Reply:
Maybe it would be even better to allow the user to set a custom ID for each credential. In this case we could just send an ID back and the user can lookup in his local setup, which credentials he actually used. To prevent sending back the credentials, we then can just disable the serialization of them with a feature flag with serde directly.